sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -535,7 +535,19 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
         |q2 as (select * from testData where key = '4')
         |select * from q1 union all select * from q2""".stripMargin),
       Row(5, "5") :: Row(4, "4") :: Nil)
+  }
 
+  test("SPARK-19799: Support WITH clause in subqueries") {
```

Comment:
Could you add more test cases? For example, can a with clause in a subquery shadow a with clause in an enclosing query with the same name? Another example, use with clauses in a subquery expression?

Reply:
I need to find some spare time for that, but I'll surely do sooner or later. I assumed that most of the correctness comes from https://github.com/apache/spark/pull/15146 but surely there may be interactions that goes beyond that.